src/marqo/tensor_search/tensor_search.py

```diff
@@ -985,6 +990,84 @@ def refresh_index(config: Config, index_name: str):
     return HttpRequests(config).post(path=F"{index_name}/_refresh")
 
 
+@add_timing
+def bulk_search(query: BulkSearchQuery, marqo_config: config.Config, reranker: Union[str, Dict] = None, return_doc_ids: bool=False, verbose: bool = True, device=None, raise_on_searchable_attribs: bool=False):
+    errs = [validation.validate_bulk_query_input(q) for q in query.queries]
+    if any(errs):
+        err = next(e for e in errs if e is not None)
+        raise err
+
+    if len(query.queries) == 0:
+        return {"result": []}
+
+    fill_index_cache(marqo_config, [q.index for q in query.queries])
```

Comment:
Index cannot be None, because Pydantic object doesn't have it as Optional
```python
class BulkSearchQueryEntity(SearchQuery):
    index: str
``` 
(if i understand pydantic correctly)